crates/cairo-lint-core/src/lints/duplicate_underscore_args.rs

```diff
@@ -0,0 +1,30 @@
+use cairo_lang_defs::plugin::PluginDiagnostic;
+use cairo_lang_semantic::Parameter;
+use cairo_lang_diagnostics::Severity;
+use std::collections::HashSet;
+// use cairo_lang_syntax::node::ids::SyntaxStablePtrId;
+
+pub const DUPLICATE_UNDERSCORE_ARGS: &str = "duplicate arguments, having another argument having almost the same name makes code comprehension and documentation more difficult";
+
+pub fn check_duplicate_underscore_args(params: Vec<Parameter>, diagnostics: &mut Vec<PluginDiagnostic>) {
+    let mut registered_names: HashSet<String> = HashSet::new();
+
+    for param in params {
+        let param_name = param.name.to_string();
+
+        if let Some(stripped_name) = param_name.strip_prefix('_') {
+            if registered_names.contains(stripped_name) {
```

Comment:
you can probably always insert and check the return value of insert https://doc.rust-lang.org/std/collections/struct.HashSet.html#method.insert